lib/transport/request.js

```diff
@@ -1,72 +1,107 @@
 'use strict';
 
-const _ = require('lodash');
-const promisifyAll = require('./promiseUtils').promisifyAll;
 const Transport = require('./transport');
-const Request = require('request');
-const dns = require('dns');
+const https = require('node:https');
+// eslint-disable-next-line func-style
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args));
 
 const REQUIRED_PROPERTIES = [
   'body',
   'elapsedTime',
   'url',
   'statusCode',
+  'statusText',
   'headers'
 ];
 
-function toAsyncMethod(method) {
-  return `${method.toLowerCase()}Async`;
-}
-
 class RequestTransport extends Transport {
-  constructor(customRequest) {
+  constructor(defaultOptions) {
     super();
-    this._request = promisifyAll(customRequest || Request);
+    if (defaultOptions?.agentOpts) {
+      this._agent = new https.Agent(defaultOptions.agentOpts);
+    }
+
+    this._fetch = fetch;
   }
 
   toOptions(ctx) {
     const req = ctx.req;
-    const opts = Object.assign({
-      time: true,
-      agentOptions: {
-        lookup: function(domain, options, cb) {
```

Comment:
Do you have any idea what the background for this might have been? Or if removing it will have any unintended consequences?

Reply:
we do not need to cover this behaviour as it is now default node behaviour  as of node 17

Reply:
https://github.com/nodejs/node/pull/39987